Common/Statistics/AssetsUnderManagementCapacityManager.cs

```diff
@@ -0,0 +1,238 @@
+
+                    // For low resolution, use the last data available
+                    if (lastData != null)
+                    {
+                        Update(sender, lastData, orderEvent.OrderId);
+                    }
+                }
+                catch (Exception e)
+                {
+                    Log.Error($"AssetsUnderManagementCapacityManager: {e}");
+                }
+            };
+        }
+
+        /// <summary>
+        /// Updates the AUM Capacity with the latest data available after a order fill
+        /// </summary>
+        /// <param name="sender">For high resolution, sender object represents the data consolidator</param>
+        /// <param name="data">Last price trade bar available or consolidated</param>
+        /// <param name="orderId">Order fill ID</param>
+        /// <remarks>
+        /// The whole method body is locked, since order events can be triggered at any time in live mode
+        /// </remarks>
+        private void Update(object sender, IBaseData data, int orderId)
+        {
+            lock (_lock)
+            {
+                try
+                {
+                    var symbol = data.Symbol;
+                    var security = _portfolio.Securities[symbol];
+
+                    var holdingsTurnover = 1m;
+                    var holdingsValue = security.Holdings.AbsoluteHoldingsValue;
+
+                    if (holdingsValue != 0)
+                    {
+                        var order = _portfolio.Transactions.GetOrderById(orderId);
+                        var orderValue = Math.Abs(order.GetValue(security));
+                        holdingsTurnover = orderValue / holdingsValue;
+
+                        if (holdingsTurnover == 0)
+                        {
+                            RemoveConsolidator(sender, symbol);
+                            return;
+                        }
+                    }
+
+                    var totalMarketVolume = GetTotalMarketVolume(data);
+                    var totalTradeVolumeCapacity = _maximumTradeableVolume * totalMarketVolume / holdingsTurnover;
+
+                    // If it is a new day, we discard the previous day data
+                    var utcDate = data.EndTime.ConvertToUtc(security.Exchange.TimeZone).Date;
+                    if (_previousInputTime != utcDate)
+                    {
+                        _previousInputTime = utcDate;
+                        _maximumSymbolCapacity.Clear();
+                        _historicalPortfolioCapacity.Add(0m);
+                    }
+
+                    // Updates AUM Capacity if there is new information or lower total trade volume capacity for the security
+                    decimal capacity;
+                    if (!_maximumSymbolCapacity.TryGetValue(symbol, out capacity) ||
+                        totalTradeVolumeCapacity < capacity)
+                    {
+                        _maximumSymbolCapacity[symbol] = totalTradeVolumeCapacity;
```

Comment:
I think the units are currently in shares not dollars. Need to multiply the `GetTotalMarketVolume` by `asset price x contract multiplier` somewhere to get AUM capacity.